sdk/src/main/java/io/radar/sdk/RadarState.kt

```diff
@@ -5,34 +5,38 @@ import android.content.SharedPreferences
 import android.location.Location
 import androidx.core.content.edit
 
-internal object RadarState {
-
-    private const val KEY_STARTED = "has_started"
-    private const val KEY_LAST_MOVED_LOCATION_LATITUDE = "last_moved_location_latitude"
-    private const val KEY_LAST_MOVED_LOCATION_LONGITUDE = "last_moved_location_longitude"
-    private const val KEY_LAST_MOVED_LOCATION_ACCURACY = "last_moved_location_accuracy"
-    private const val KEY_LAST_MOVED_LOCATION_PROVIDER = "last_moved_location_provider"
-    private const val KEY_LAST_MOVED_LOCATION_TIME = "last_moved_location_time"
-    private const val KEY_LAST_MOVED_AT = "last_moved_at"
-    private const val KEY_STOPPED = "stopped"
-    private const val KEY_LAST_SENT_AT = "last_sent_at"
-    private const val KEY_CAN_EXIT = "can_exit"
-    private const val KEY_LAST_FAILED_STOPPED_LOCATION_LATITUDE = "last_failed_stopped_location_latitude"
-    private const val KEY_LAST_FAILED_STOPPED_LOCATION_LONGITUDE = "last_failed_stopped_location_longitude"
-    private const val KEY_LAST_FAILED_STOPPED_LOCATION_ACCURACY = "last_failed_stopped_location_accuracy"
-    private const val KEY_LAST_FAILED_STOPPED_LOCATION_PROVIDER = "last_failed_stopped_location_provider"
-    private const val KEY_LAST_FAILED_STOPPED_LOCATION_TIME = "last_failed_stopped_location_time"
-
-    private fun getSharedPreferences(context: Context): SharedPreferences {
-        return context.getSharedPreferences("RadarSDK", Context.MODE_PRIVATE)
+@Suppress("TooManyFunctions")
+internal class RadarState(private val context: Context) {
```

Comment:
I can see the appeal of not having to pass in a context for testability to mock a context, but I'm not quite sure how moving it to the constructor has helped apart from reducing repetition of passing in context, which is certainly nice.

Reply:
This is all part of a significantly larger refactor. RadarState becomes a field of a new object that is used for dependency injection and yes, mocking things out becomes easy, since there are no static methods used anymore.